kafkaserver/handler.go

```diff
@@ -75,7 +75,23 @@ func (n *NewHandler) HandleProduceRequest(_ *protocol.RequestHeader, req *protoc
 				cf.CountDown(nil)
 				continue
 			}
+
 			producedRecords := partitionData.Records[0]
+			producerID := int(binary.BigEndian.Uint64(producedRecords[43:51]))
```

Comment:
I think we should move this chunk of code to KafkaInOperator.HandleStreamBatch - this is called on the eventloop with the batch. Then you can extract the producer id and sequence there and lookup from the map stored there